rust-runtime/aws-smithy-types/src/number.rs

```diff
@@ -3,11 +3,34 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+//! A number type that implements Javascript / JSON semantics, modeled on serde_json:
+//! <https://docs.serde.rs/src/serde_json/number.rs.html#20-22>
```

Comment:
To keep the summary line for the module shorter:
```suggestion
//! A number type that implements Javascript / JSON semantics.
```